game/theater/start_generator.py

```diff
@@ -161,38 +142,79 @@ def faction_name(self) -> str:
     def faction(self) -> Faction:
         return self.game.coalition_for(self.control_point.captured).faction
 
+    @property
+    def ground_forces(self) -> GroundForces:
+        return self.game.coalition_for(self.control_point.captured).ground_forces
+
     def generate(self) -> bool:
         self.control_point.connected_objectives = []
-        if self.faction.navy_generators:
-            # Even airbases can generate navies if they are close enough to the water.
-            self.generate_navy()
-
+        self.generate_navy()
         return True
 
+    def generate_ground_object_for_preset(
+        self, preset_location: PresetLocation
+    ) -> None:
+        """Generate a ground_object for a given PresetLocation"""
+        if preset_location.task is None:
+            logging.error(
+                f"PresetLocation {preset_location.original_name} has no task. Skipping"
+            )
+            return
```

Comment:
I agree on the non-optional property because it should never be None.
As suggested i will move it to a different PR but my Idea behind it is rather simple: Everything has a specific tasking (Role can be removed as it will be redundant when Task is required and a subclass of GroupRole as already mentionend in previous comments by you)
And regarding the original_task property: It is only used during the start_generator so only once to initally create a TGO from the Campaign Designer placed units. Will not be used again later IIRC. But ofc i will check that within the next days :)